shared/domains/chelonia/chelonia.js

```diff
@@ -673,7 +673,7 @@ export default (sbp('sbp/selectors/register', {
     const payload = ({
       keyId: innerSigningKeyId,
       encryptionKeyId: encryptionKeyId,
-      data: JSON.stringify(signatureFnBuilder(innerSigningKey)(`${originatingContractID}${GIMessage.OP_KEY_REQUEST}${contractID}${previousHEAD}`))
+      data: sign(innerSigningKey, [originatingContractID, GIMessage.OP_KEY_REQUEST, contractID, previousHEAD].map(encodeURIComponent).join('|'))
```

Comment:
Is it necessary for some reason to map `encodeURIComponent`? Or can you simply call `.join`?

Reply:
The reason in this case is so that the separator is escaped. An alternative is to escape just the separator.

Reply:
Wait, why is anything being escaped?

Reply:
Because not escaping could make things ambiguous (not in this particular case considering the character set the current implementation uses for these values, but that's relying on internals that could change in the future).

`[originatingContractID, GIMessage.OP_KEY_REQUEST, contractID, previousHEAD]`

Imagine that `contractID = 'a|b'` and `previousHEAD = 'c'`. Then the concat function would make it undistinguishable from `contractID = 'a'` and `previousHEAD = 'b|c'` without escaping.

Reply:
But `|` will never appear inside of a `contractID` or a `previousHEAD`, that we know, and it's not part of the protocol for that to change...

Reply:
That's true, but it's one of the things that could lead to unpredictably broken code in the future if it should change. Escaping seems safer and doesn't require making assumptions about internal data structures that may or may not change, for a probably negligible cost.

Reply:
Summary of call: we agreed that since not all languages implement `encodeURIComponent`, we would simply check if any of the variables in `data` array contain `|` we'd throw an exception. Feel free to mark resolved once changes are pushed.